packages/xo-server-audit/src/index.js

```diff
@@ -0,0 +1,159 @@
+import appConf from 'app-conf'
+import asyncIteratorToStream from 'async-iterator-to-stream'
+import createLogger from '@xen-orchestra/log'
+import path from 'path'
+import { AuditCore, Storage } from '@xen-orchestra/audit-core'
+import { fromCallback } from 'promise-toolbox'
+import { pipeline } from 'readable-stream'
+
+const log = createLogger('xo:xo-server-audit')
+
+const LAST_ID = 'lastId'
+class DB extends Storage {
+  constructor(db) {
+    super()
+    this._db = db
+  }
+
+  async put(record) {
+    await this._db.put(record.id, record)
+  }
+
+  async del(id) {
+    await this._db.del(id)
+  }
+
+  get(id) {
+    return this._db.get(id).catch(error => {
+      if (!error.notFound) {
+        throw error
+      }
+    })
+  }
+
+  async setLastId(id) {
+    await this._db.put(LAST_ID, id)
+  }
+
+  getLastId() {
+    return this.get(LAST_ID)
+  }
+}
+
+const NAMESPACE = 'audit'
+class AuditXoPlugin {
+  constructor({ xo }) {
+    this._cleaners = []
+    this._xo = xo
+
+    this._auditCore = undefined
+    this._blockList = undefined
+  }
+
+  async load() {
+    const cleaners = this._cleaners
+
+    try {
+      this._auditCore = new AuditCore(
+        new DB(await this._xo.getStore(NAMESPACE))
+      )
+      this._blockList = (
+        await appConf.load('xo-server-audit', {
+          appDir: path.join(__dirname, '..'),
+        })
+      ).blockList
+
+      cleaners.push(() => {
+        this._auditCore = undefined
+        this._blockList = undefined
+      })
```

Comment:
Don't need to store them in `cleaners`.

Reply:
The plugin cache should be totally clean in case of the unload.

Reply:
Yes, I mean move it to `unload` instead of creating a function here.

Reply:
The benefits of this generic collection is to keep allowing/clearing properties in the same place which will allow us to not forget the clean of a property and improve readability. The load/unload of the plugin will not be executed multiple times which will not impact the plugin performance.